app/controllers/ideas_controller.rb

```diff
@@ -73,6 +74,20 @@ def set_idea
     @idea = Idea.find(params[:id] || params[:idea_id])
   end
 
+  def update_review_date(idea_params)
+    params_to_scrub = %i[review_year review_month review_day]
+    new_params = idea_params.except(*params_to_scrub)
+    if idea_params[:review_year].present? || idea_params[:review_month].present? || idea_params[:review_day].present?
+      begin
+        date_str = "#{idea_params[:review_year]}-#{idea_params[:review_month]}-#{idea_params[:review_day]}"
```

Comment:
Doing this in the model would make sense. You could use a `before_validate` callback to achieve this, similar to how it is (or at least was) done on Apply. Each of the 3 fields will need to be declared as writable attributes on the model (but not db attributes) - you have made them readable (but from the db).